test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe(
+	'As a user I want the navigation block to fallback to the best possible default',
+	() => {
+		test.beforeEach( async ( { admin } ) => {
+			await admin.createNewPost();
+		} );
+
+		test( 'default to a list of pages if there are no menus', async ( {
+			editor,
+		} ) => {
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			// Check Page List is in the list view.
+
+			// Open the list view.
+			await editor.page.locator( '[aria-label="List View"i]' ).click();
+			// Click the Navigation block expander, we need to use force because it has aria-hidden set to true.
+			await editor.page
+				.locator(
+					`//a[.//span[text()='Navigation']]/span[contains(@class, 'block-editor-list-view__expander')]`
+				)
+				.click( { force: true } );
+			// Find the Page list selector inside the navigation block.
+			const pageListSelector = await editor.page.locator(
+				`//table[contains(@aria-label,'Block navigation structure')]//a[.//span[text()='Page List']]`
+			);
+
+			expect( pageListSelector ).toBeTruthy();
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			const content = await editor.getEditedPostContent();
+			expect( content ).toBe(
+				`<!-- wp:navigation -->
+<!-- wp:page-list /-->
+<!-- /wp:navigation -->`
+			);
+		} );
+
+		test( 'default to my only existing menu', async () => {} );
+		test( 'default to my most recently created menu', async () => {} );
+		test( 'default to the only existing classic menu if there are no block menus', async () => {} );
+	}
+);
+
+test.describe(
+	'As a user I want to see a warning if the menu referenced by a navigation block is not available',
+	() => {
+		test.beforeEach( async ( { admin } ) => {
+			await admin.createNewPost();
+		} );
+
+		test( 'warning message shows when given an unknown ref', async ( {
+			editor,
+		} ) => {
+			await editor.insertBlock( {
+				name: 'core/navigation',
+				attributes: {
+					ref: 1,
+				},
+			} );
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			const content = await editor.getEditedPostContent();
+			expect( content ).toBe( `<!-- wp:navigation {"ref":1} /-->` );
+
+			// Find the warning message
+			const warningMessage = await editor.page.locator(
+				'class=wp-navigation-block',
```

Comment:
It's be great to use role selectors where possible over class names—it helps ensure the UI has accessible markup.

For the block it'd be `role=document[name="Block: Navigation"i]` (the document role for blocks is a bit unusual, but that was apparently necessary). The easiest way to get the right values is to inspect the element in the browser and check the accessibility tab's computed properties.

There's also a shorter way of selecting text `role=document[name="Block: Navigation"i] >> text="Navigation menu has been deleted or is unavailable."`

I think you should be able to shorten the whole thing to this:
```js
await expect( `role=document[name="Block: Navigation"i] >> text="Navigation menu has been deleted or is unavailable." ).toBeVisible();
```

Reply:
Everything @talldan said 😆, plus a couple of things:

1. You can use `page` directly rather than referencing `editor.page`. `editor.page` isn't supposed to be a public API and might be marked private in the future.
2. Locator doesn't return a promise, so no need to `await` it.
3. Playwright deletes the doc for role selectors 😞 , but thankfully the alternative `getByRole` is basically the same thing but more friendly if you're familiar with testing-library's API.


Hence, the whole thing could be:

```js
const warningMessage = page.getByRole( 'document', { name: 'Block: Navigation' } )
	.getByText( 'Navigation menu has been deleted or is unavailable.' );
```

Reply:
`page` didn't work (` ReferenceError: page is not defined`) but `editor.canvas` does, which is what we use in other other test.

Reply:
In order to use `page` you'll have to specify it in the test callback parameter: 

```js
test( '...', async ( { page } ) => {
```

But yeah, since that now post editor is rendered inside an iframe we can just use `editor.canvas` instead.